
 <!DOCTYPE HTML>
<html lang="zh-Hans">
<head><meta name="generator" content="Hexo 3.8.0">
  <meta charset="UTF-8">
  
    <title>procrustes分析 | 有味</title>
    <meta name="viewport" content="width=device-width, initial-scale=1,user-scalable=no">
    
    <meta name="author" content="Dayueban">
    

    
    <meta name="description" content="目的：为了展示相同个体不同组学数据，或者不同表型数据之间的相似性关系以及之间的差异性，所以采用了普氏分析。这里展示的是如何将两组数据进行普氏分析并进行可视化 一、导入数据1 | 微生物种水平丰度表">
<meta name="keywords" content="R,普氏分析">
<meta property="og:type" content="article">
<meta property="og:title" content="procrustes分析">
<meta property="og:url" content="https://dayueban.com/2019/06/15/procrustes分析/index.html">
<meta property="og:site_name" content="有味">
<meta property="og:description" content="目的：为了展示相同个体不同组学数据，或者不同表型数据之间的相似性关系以及之间的差异性，所以采用了普氏分析。这里展示的是如何将两组数据进行普氏分析并进行可视化 一、导入数据1 | 微生物种水平丰度表">
<meta property="og:locale" content="zh-Hans">
<meta property="og:image" content="https://s2.ax1x.com/2019/06/15/Vonrp8.png">
<meta property="og:image" content="https://s2.ax1x.com/2019/06/15/VoupcD.png">
<meta property="og:image" content="https://s2.ax1x.com/2019/06/15/VoQ8UA.png">
<meta property="og:updated_time" content="2019-06-15T10:30:54.986Z">
<meta name="twitter:card" content="summary">
<meta name="twitter:title" content="procrustes分析">
<meta name="twitter:description" content="目的：为了展示相同个体不同组学数据，或者不同表型数据之间的相似性关系以及之间的差异性，所以采用了普氏分析。这里展示的是如何将两组数据进行普氏分析并进行可视化 一、导入数据1 | 微生物种水平丰度表">
<meta name="twitter:image" content="https://s2.ax1x.com/2019/06/15/Vonrp8.png">
<meta name="twitter:creator" content="@https:&#x2F;&#x2F;twitter.com&#x2F;HeMaozhang">

    
    <link rel="alternative" href="/atom.xml" title="有味" type="application/atom+xml">
    
    
    <link rel="icon" href="/img/favicon1.ico">
    
    
    <link rel="apple-touch-icon" href="/img/jacman1.jpg">
    <link rel="apple-touch-icon-precomposed" href="/img/jacman1.jpg">
    
    <link rel="stylesheet" href="/css/style.css">
</head>
</html>
  <body>
    <header>
      
<div>
		
			<div id="imglogo">
				<a href="/"><img src="/img/logo1.png" alt="有味" title="有味"></a>
			</div>
			
			<div id="textlogo">
				<h1 class="site-name"><a href="/" title="有味">有味</a></h1>
				<h2 class="blog-motto">当悦纳自我时，每一刻都是最好的自己！</h2>
			</div>
			<div class="navbar"><a class="navbutton navmobile" href="#" title="Menu">
			</a></div>
			<nav class="animated">
				<ul>
					<ul>
					 
						<li><a href="/">主页 | Home</a></li>
					
						<li><a href="/archives">文件夹 | Archives</a></li>
					
						<li><a href="/about">关于我 | About me</a></li>
					
					<li>
 					
						<form class="search" action="http://zhannei.baidu.com/cse/search" target="_blank">
							<label>Search</label>
						<input name="s" type="hidden" value=""><input type="text" name="q" size="30" placeholder="Search"><br>
						</form>
					
					</li>
				</ul>
			</ul></nav>			
</div>
    </header>
    <div id="container">
      <div id="main" class="post" itemscope="" itemprop="blogPost">
  
	<article itemprop="articleBody"> 
		<header class="article-info clearfix">
  <h1 itemprop="name">
    
      <a href="/2019/06/15/procrustes分析/" title="procrustes分析" itemprop="url">procrustes分析</a>
  </h1>
  <p class="article-author">By
       
		<a href="/about" title="Dayueban" target="_blank" itemprop="author">Dayueban</a>
		
  </p><p class="article-time">
    <time datetime="2019-06-15T09:39:57.000Z" itemprop="datePublished"> Published 2019-06-15</time>
    
  </p>
</header>
	<div class="article-content">
		
		<div id="toc" class="toc-article">
			<strong class="toc-title">Contents</strong>
		
			<ol class="toc"><li class="toc-item toc-level-2"><a class="toc-link" href="#目的："><span class="toc-number">1.</span> <span class="toc-text">目的：</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#一、导入数据"><span class="toc-number">2.</span> <span class="toc-text">一、导入数据</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#二、进行procrustes分析并作图"><span class="toc-number">3.</span> <span class="toc-text">二、进行procrustes分析并作图</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#参考文章"><span class="toc-number">4.</span> <span class="toc-text">参考文章</span></a></li></ol>
		
		</div>
		
		<h2 id="目的："><a href="#目的：" class="headerlink" title="目的："></a>目的：</h2><p>为了展示相同个体不同组学数据，或者不同表型数据之间的相似性关系以及之间的差异性，所以采用了普氏分析。这里展示的是如何将两组数据进行普氏分析并进行可视化</p>
<h2 id="一、导入数据"><a href="#一、导入数据" class="headerlink" title="一、导入数据"></a>一、导入数据</h2><p>1 | 微生物种水平丰度表</p>
<p><img src="https://s2.ax1x.com/2019/06/15/Vonrp8.png" alt="图一 微生物种水平丰度表"></p>
<a id="more"></a>
<p>2 | 食物数据的非加权unifrac矩阵</p>
<p><img src="https://s2.ax1x.com/2019/06/15/VoupcD.png" alt="图二 食物数据"></p>
<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment"># 食物数据</span></span><br><span class="line"></span><br><span class="line"><span class="comment"># load the food distance matrix, unweighted unifrac</span></span><br><span class="line">food_un &lt;- read.delim(<span class="string">"data/diet/processed_food/dhydrt_smry_no_soy_beta/unweighted_unifrac_dhydrt.smry.no.soy.txt"</span>, row = <span class="number">1</span>) <span class="comment"># weighted in not significant</span></span><br><span class="line">food_dist &lt;- as.dist(food_un)</span><br><span class="line"></span><br><span class="line"><span class="comment">##############taxonomy############</span></span><br><span class="line"><span class="comment"># load taxonomy collapsed for each person</span></span><br><span class="line">tax &lt;- read.delim(<span class="string">"data/microbiome/processed_average/UN_tax_CLR_mean_norm_s.txt"</span>, row = <span class="number">1</span>) <span class="comment"># updates from reviewer comments</span></span><br><span class="line"><span class="comment"># drop soylents</span></span><br><span class="line">tax &lt;- tax[,colnames(tax) %<span class="keyword">in</span>% colnames(food_un)]</span><br><span class="line"></span><br><span class="line"><span class="comment"># make taxonomy distance matrix</span></span><br><span class="line">tax_dist &lt;- dist(t(tax))</span><br><span class="line"></span><br><span class="line"><span class="comment">###############nutrients############</span></span><br><span class="line"><span class="comment"># load nutrition data</span></span><br><span class="line">nutr &lt;- read.delim(<span class="string">"data/diet/processed_nutr/nutr_65_smry_no_soy.txt"</span>, row = <span class="number">1</span>)</span><br><span class="line"></span><br><span class="line"><span class="comment"># normalize nutrition data across features (rows)</span></span><br><span class="line">nutr_n &lt;- sweep(nutr, <span class="number">1</span>, rowSums(nutr), <span class="string">"/"</span>)</span><br><span class="line"></span><br><span class="line"><span class="comment"># make nutrition distance matrix (euclidean)</span></span><br><span class="line">nutr_dist &lt;- dist(t(nutr_n))</span><br></pre></td></tr></table></figure>
<h2 id="二、进行procrustes分析并作图"><a href="#二、进行procrustes分析并作图" class="headerlink" title="二、进行procrustes分析并作图"></a>二、进行procrustes分析并作图</h2><p>1 | 首先以食物和微生物为例<br><figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br><span class="line">26</span><br><span class="line">27</span><br><span class="line">28</span><br><span class="line">29</span><br><span class="line">30</span><br><span class="line">31</span><br><span class="line">32</span><br><span class="line">33</span><br><span class="line">34</span><br><span class="line">35</span><br><span class="line">36</span><br><span class="line">37</span><br><span class="line">38</span><br><span class="line">39</span><br><span class="line">40</span><br><span class="line">41</span><br><span class="line">42</span><br><span class="line">43</span><br><span class="line">44</span><br><span class="line">45</span><br><span class="line">46</span><br><span class="line">47</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment"># make pcoas </span></span><br><span class="line">pcoa_f &lt;- as.data.frame(pcoa(food_dist)$vectors)</span><br><span class="line">pcoa_t &lt;- as.data.frame(pcoa(tax_dist)$vectors)</span><br><span class="line"></span><br><span class="line"><span class="comment"># procrustes</span></span><br><span class="line">pro &lt;- procrustes(pcoa_f, pcoa_t)</span><br><span class="line">pro_test &lt;- protest(pcoa_f, pcoa_t, perm = <span class="number">999</span>)  <span class="comment">#普氏分析组间数据的检验</span></span><br><span class="line"></span><br><span class="line">eigen &lt;- sqrt(pro$svd$d)</span><br><span class="line">percent_var &lt;- signif(eigen/sum(eigen), <span class="number">4</span>)*<span class="number">100</span></span><br><span class="line"></span><br><span class="line">beta_pro &lt;- data.frame(pro$X)</span><br><span class="line">trans_pro &lt;- data.frame(pro$Yrot)</span><br><span class="line">beta_pro$UserName &lt;- rownames(beta_pro)</span><br><span class="line">beta_pro$type &lt;- <span class="string">"Food (Unweighted Unifrac)"</span></span><br><span class="line">trans_pro$UserName &lt;- rownames(trans_pro)</span><br><span class="line">trans_pro$type &lt;- <span class="string">"Microbiome (Aitchison's)"</span></span><br><span class="line"></span><br><span class="line">colnames(trans_pro) &lt;- colnames(beta_pro)</span><br><span class="line"></span><br><span class="line">pval &lt;- signif(pro_test$signif, <span class="number">1</span>)</span><br><span class="line"></span><br><span class="line">plot &lt;- rbind(beta_pro, trans_pro)</span><br><span class="line"></span><br><span class="line">food_micro &lt;- ggplot(plot) +</span><br><span class="line">  geom_point(size = <span class="number">2</span>, alpha=<span class="number">0.75</span>, aes(x = Axis.1, y = Axis.2, color = type)) +</span><br><span class="line">  scale_color_manual(values = c(<span class="string">"#5a2071"</span>, <span class="string">"#5f86b7"</span>)) +</span><br><span class="line">    theme_classic() +</span><br><span class="line">    geom_line(aes(x= Axis.1, y=Axis.2, group=UserName), col = <span class="string">"darkgrey"</span>, alpha = <span class="number">0.6</span>) +</span><br><span class="line">  theme(panel.background = element_blank(),</span><br><span class="line">        panel.grid.major = element_blank(),</span><br><span class="line">        panel.grid.minor = element_blank(),</span><br><span class="line">        legend.title = element_blank(),</span><br><span class="line">        legend.text = element_text(size=<span class="number">9</span>),</span><br><span class="line">        legend.position = <span class="string">'bottom'</span>,</span><br><span class="line">        axis.text = element_text(size=<span class="number">4</span>),</span><br><span class="line">        axis.title = element_text(size=<span class="number">9</span>),</span><br><span class="line">         aspect.ratio = <span class="number">1</span>) +</span><br><span class="line">  guides(color = guide_legend(ncol = <span class="number">1</span>)) +</span><br><span class="line">  annotate(<span class="string">"text"</span>, x = <span class="number">0.3</span>, y = -<span class="number">0.27</span>, label = paste0(<span class="string">"p-value="</span>,pval), size = <span class="number">2</span>) +</span><br><span class="line">  xlab(paste0(<span class="string">"PC 1 ["</span>,percent_var[<span class="number">1</span>],<span class="string">"%]"</span>)) +</span><br><span class="line">  ylab(paste0(<span class="string">"PC 2 ["</span>,percent_var[<span class="number">2</span>],<span class="string">"%]"</span>)) </span><br><span class="line">  </span><br><span class="line">  food_micro_leg &lt;- get_legend(food_micro) <span class="comment">#得到ggplot图的图例信息</span></span><br><span class="line"></span><br><span class="line"></span><br><span class="line">food_micro + theme(legend.position = <span class="string">"none"</span>)</span><br></pre></td></tr></table></figure></p>
<p>2 | 营养与微生物之间的普氏分析<br><figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br><span class="line">26</span><br><span class="line">27</span><br><span class="line">28</span><br><span class="line">29</span><br><span class="line">30</span><br><span class="line">31</span><br><span class="line">32</span><br><span class="line">33</span><br><span class="line">34</span><br><span class="line">35</span><br><span class="line">36</span><br><span class="line">37</span><br><span class="line">38</span><br><span class="line">39</span><br><span class="line">40</span><br><span class="line">41</span><br><span class="line">42</span><br><span class="line">43</span><br><span class="line">44</span><br><span class="line">45</span><br><span class="line">46</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment"># make pcoas </span></span><br><span class="line">pcoa_n &lt;- as.data.frame(pcoa(nutr_dist)$vectors)</span><br><span class="line">pcoa_t &lt;- as.data.frame(pcoa(tax_dist)$vectors)</span><br><span class="line"></span><br><span class="line"><span class="comment"># procrustes</span></span><br><span class="line">pro &lt;- procrustes(pcoa_n, pcoa_t)</span><br><span class="line">pro_test &lt;- protest(pcoa_n, pcoa_t, perm = <span class="number">999</span>)</span><br><span class="line"></span><br><span class="line">eigen &lt;- sqrt(pro$svd$d)</span><br><span class="line">percent_var &lt;- signif(eigen/sum(eigen), <span class="number">4</span>)*<span class="number">100</span></span><br><span class="line"></span><br><span class="line">beta_pro &lt;- data.frame(pro$X)</span><br><span class="line">trans_pro &lt;- data.frame(pro$Yrot)</span><br><span class="line">beta_pro$UserName &lt;- rownames(beta_pro)</span><br><span class="line">beta_pro$type &lt;- <span class="string">"Nutrient (Euclidean)"</span></span><br><span class="line">trans_pro$UserName &lt;- rownames(trans_pro)</span><br><span class="line">trans_pro$type &lt;- <span class="string">"Microbiome (Aitchison's)"</span></span><br><span class="line"></span><br><span class="line">colnames(trans_pro) &lt;- colnames(beta_pro)</span><br><span class="line"></span><br><span class="line">pval &lt;- signif(pro_test$signif, <span class="number">1</span>)</span><br><span class="line"></span><br><span class="line">plot &lt;- rbind(beta_pro, trans_pro)</span><br><span class="line"></span><br><span class="line">nutr_micro &lt;- ggplot(plot) +</span><br><span class="line">  geom_point(size = <span class="number">2</span>, alpha=<span class="number">0.75</span>, aes(x = Axis.1, y = Axis.2, color = type)) +</span><br><span class="line">  scale_color_manual(values = c(<span class="string">"#5f86b7"</span>, <span class="string">"#2bbaa7"</span>)) +</span><br><span class="line">    theme_classic() +</span><br><span class="line">    geom_line(aes(x= Axis.1, y=Axis.2, group=UserName), col = <span class="string">"darkgrey"</span>, alpha = <span class="number">0.6</span>) +</span><br><span class="line">  theme(panel.background = element_blank(),</span><br><span class="line">        panel.grid.major = element_blank(),</span><br><span class="line">        panel.grid.minor = element_blank(),</span><br><span class="line">        legend.title = element_blank(),</span><br><span class="line">        legend.text = element_text(size=<span class="number">9</span>),</span><br><span class="line">        legend.position = <span class="string">'bottom'</span>,</span><br><span class="line">        axis.text = element_text(size=<span class="number">4</span>),</span><br><span class="line">        axis.title = element_text(size=<span class="number">9</span>),</span><br><span class="line">         aspect.ratio = <span class="number">1</span>) +</span><br><span class="line">  guides(color = guide_legend(ncol = <span class="number">1</span>)) +</span><br><span class="line">  annotate(<span class="string">"text"</span>, x = <span class="number">0.01</span>, y = -<span class="number">0.19</span>, label = paste0(<span class="string">"p-value="</span>,pval), size = <span class="number">2</span>) +</span><br><span class="line">  xlab(paste0(<span class="string">"PC 1 ["</span>,percent_var[<span class="number">1</span>],<span class="string">"%]"</span>)) +</span><br><span class="line">  ylab(paste0(<span class="string">"PC 2 ["</span>,percent_var[<span class="number">2</span>],<span class="string">"%]"</span>)) </span><br><span class="line"></span><br><span class="line">nutr_micro + theme(legend.position = <span class="string">"none"</span>)</span><br><span class="line"></span><br><span class="line">nutr_micro_leg &lt;- get_legend(nutr_micro)</span><br></pre></td></tr></table></figure></p>
<p>3 | 谷物类数据和微生物数据（欧几里得距离）<br><figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br><span class="line">26</span><br><span class="line">27</span><br><span class="line">28</span><br><span class="line">29</span><br><span class="line">30</span><br><span class="line">31</span><br><span class="line">32</span><br><span class="line">33</span><br><span class="line">34</span><br><span class="line">35</span><br><span class="line">36</span><br><span class="line">37</span><br><span class="line">38</span><br><span class="line">39</span><br><span class="line">40</span><br><span class="line">41</span><br><span class="line">42</span><br><span class="line">43</span><br><span class="line">44</span><br><span class="line">45</span><br><span class="line">46</span><br><span class="line">47</span><br><span class="line">48</span><br><span class="line">49</span><br><span class="line">50</span><br><span class="line">51</span><br><span class="line">52</span><br><span class="line">53</span><br><span class="line">54</span><br><span class="line">55</span><br><span class="line">56</span><br><span class="line">57</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment"># Import the beta diversity table from grains_beta (weighted or unweighted)</span></span><br><span class="line">food_beta &lt;- read.table(file=<span class="string">"data/diet/fiber/grains_data/grains_beta/unweighted_unifrac_grains_fiber.txt"</span>)</span><br><span class="line"></span><br><span class="line">food_dist &lt;- as.dist(food_beta)</span><br><span class="line"><span class="comment"># refer this code to make a pcoa you will have to fix naming and play with the data structure of the dataframe called plot to get this to work.</span></span><br><span class="line"></span><br><span class="line"><span class="comment"># Import the beta diversity table for taxonomy</span></span><br><span class="line">taxa_beta &lt;- read.table(file=<span class="string">"data/microbiome/processed_average/UN_tax_beta/euclidean_UN_taxonomy_clr_s.txt"</span>)</span><br><span class="line"></span><br><span class="line">tax_dist &lt;- as.dist(taxa_beta)</span><br><span class="line"></span><br><span class="line"><span class="comment"># make pcoas </span></span><br><span class="line">pcoa_f &lt;- as.data.frame(pcoa(food_dist)$vectors)</span><br><span class="line">pcoa_t &lt;- as.data.frame(pcoa(tax_dist)$vectors)</span><br><span class="line"></span><br><span class="line"><span class="comment"># procrustes</span></span><br><span class="line">pro &lt;- procrustes(pcoa_f, pcoa_t)</span><br><span class="line">pro_test &lt;- protest(pcoa_f, pcoa_t, perm = <span class="number">999</span>)</span><br><span class="line"></span><br><span class="line">eigen &lt;- sqrt(pro$svd$d)</span><br><span class="line">percent_var &lt;- signif(eigen/sum(eigen), <span class="number">4</span>)*<span class="number">100</span></span><br><span class="line"></span><br><span class="line">beta_pro &lt;- data.frame(pro$X)</span><br><span class="line">trans_pro &lt;- data.frame(pro$Yrot)</span><br><span class="line">beta_pro$UserName &lt;- rownames(beta_pro)</span><br><span class="line">beta_pro$type &lt;- <span class="string">"Grain Fiber (Unweighted Unifrac)"</span></span><br><span class="line">trans_pro$UserName &lt;- rownames(trans_pro)</span><br><span class="line">trans_pro$type &lt;- <span class="string">"Microbiome (Aitchison's)"</span></span><br><span class="line"></span><br><span class="line">colnames(trans_pro) &lt;- colnames(beta_pro)</span><br><span class="line"></span><br><span class="line">pval &lt;- signif(pro_test$signif)</span><br><span class="line"></span><br><span class="line">plot &lt;- rbind(beta_pro, trans_pro)</span><br><span class="line"></span><br><span class="line">grain_micro &lt;- ggplot(plot) +</span><br><span class="line">  geom_point(size = <span class="number">2</span>, alpha=<span class="number">0.75</span>, aes(x = Axis.1, y = Axis.2, color = type)) +</span><br><span class="line">  scale_color_manual(values = c(<span class="string">"#fe9700"</span>, <span class="string">"#5f86b7"</span>)) +</span><br><span class="line">    theme_classic() +</span><br><span class="line">    geom_line(aes(x= Axis.1, y=Axis.2, group=UserName), col = <span class="string">"darkgrey"</span>, alpha = <span class="number">0.6</span>) +</span><br><span class="line">  theme(panel.background = element_blank(),</span><br><span class="line">        panel.grid.major = element_blank(),</span><br><span class="line">        panel.grid.minor = element_blank(),</span><br><span class="line">        legend.title = element_blank(),</span><br><span class="line">        legend.text = element_text(size=<span class="number">9</span>),</span><br><span class="line">        legend.position = <span class="string">'bottom'</span>,</span><br><span class="line">        axis.text = element_text(size=<span class="number">4</span>),</span><br><span class="line">        axis.title = element_text(size=<span class="number">9</span>),</span><br><span class="line">         aspect.ratio = <span class="number">1</span>) +</span><br><span class="line">  guides(color = guide_legend(ncol = <span class="number">1</span>)) +</span><br><span class="line">  annotate(<span class="string">"text"</span>, x = <span class="number">0.05</span>, y = -<span class="number">0.27</span>, label = paste0(<span class="string">"p-value="</span>,pval), size = <span class="number">2</span>) +</span><br><span class="line">  xlab(paste0(<span class="string">"PC 1 ["</span>,percent_var[<span class="number">1</span>],<span class="string">"%]"</span>)) +</span><br><span class="line">  ylab(paste0(<span class="string">"PC 2 ["</span>,percent_var[<span class="number">2</span>],<span class="string">"%]"</span>)) </span><br><span class="line"></span><br><span class="line">grain_micro + theme(legend.position = <span class="string">"none"</span>)</span><br><span class="line"></span><br><span class="line">grain_leg &lt;- get_legend(grain_micro)</span><br></pre></td></tr></table></figure></p>
<p>4 | 水果数据和微生物数据<br><figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br><span class="line">26</span><br><span class="line">27</span><br><span class="line">28</span><br><span class="line">29</span><br><span class="line">30</span><br><span class="line">31</span><br><span class="line">32</span><br><span class="line">33</span><br><span class="line">34</span><br><span class="line">35</span><br><span class="line">36</span><br><span class="line">37</span><br><span class="line">38</span><br><span class="line">39</span><br><span class="line">40</span><br><span class="line">41</span><br><span class="line">42</span><br><span class="line">43</span><br><span class="line">44</span><br><span class="line">45</span><br><span class="line">46</span><br><span class="line">47</span><br><span class="line">48</span><br><span class="line">49</span><br><span class="line">50</span><br><span class="line">51</span><br><span class="line">52</span><br><span class="line">53</span><br><span class="line">54</span><br><span class="line">55</span><br><span class="line">56</span><br><span class="line">57</span><br><span class="line">58</span><br><span class="line">59</span><br><span class="line">60</span><br><span class="line">61</span><br><span class="line">62</span><br><span class="line">63</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment"># Import the beta diversity table from fruit_beta (weighted or unweighted)</span></span><br><span class="line">food_beta &lt;- read.table(file=<span class="string">"data/diet/fiber/fruit_data/fruit_beta/unweighted_unifrac_fruit_fiber.txt"</span>)</span><br><span class="line"></span><br><span class="line">food_dist &lt;- as.dist(food_beta)</span><br><span class="line"></span><br><span class="line"><span class="comment"># make pcoas </span></span><br><span class="line">pcoa_f &lt;- as.data.frame(pcoa(food_dist)$vectors)</span><br><span class="line">pcoa_t &lt;- as.data.frame(pcoa(tax_dist)$vectors)</span><br><span class="line"></span><br><span class="line"><span class="comment"># procrustes</span></span><br><span class="line">pro &lt;- procrustes(pcoa_f, pcoa_t)</span><br><span class="line">pro_test &lt;- protest(pcoa_f, pcoa_t, perm = <span class="number">999</span>)</span><br><span class="line"></span><br><span class="line">eigen &lt;- sqrt(pro$svd$d)</span><br><span class="line">percent_var &lt;- signif(eigen/sum(eigen), <span class="number">4</span>)*<span class="number">100</span></span><br><span class="line"></span><br><span class="line">beta_pro &lt;- data.frame(pro$X)</span><br><span class="line">trans_pro &lt;- data.frame(pro$Yrot)</span><br><span class="line">beta_pro$UserName &lt;- rownames(beta_pro)</span><br><span class="line">beta_pro$type &lt;- <span class="string">"Fruit Fiber (Unweighted Unifrac)"</span></span><br><span class="line">trans_pro$UserName &lt;- rownames(trans_pro)</span><br><span class="line">trans_pro$type &lt;- <span class="string">"Microbiome (CLR-Euclidean)"</span></span><br><span class="line"></span><br><span class="line">colnames(trans_pro) &lt;- colnames(beta_pro)</span><br><span class="line"></span><br><span class="line">pval &lt;- signif(pro_test$signif, <span class="number">3</span>)</span><br><span class="line"></span><br><span class="line">plot &lt;- rbind(beta_pro, trans_pro)</span><br><span class="line"></span><br><span class="line">fruit_micro &lt;- ggplot(plot) +</span><br><span class="line">    geom_point(size = <span class="number">2</span>, alpha=<span class="number">0.75</span>, aes(x = Axis.1, y = Axis.2, color = type)) +</span><br><span class="line">  scale_color_manual(values = c(<span class="string">"#CBD13E"</span>, <span class="string">"#5f86b7"</span>)) +</span><br><span class="line">    theme_classic() +</span><br><span class="line">    geom_line(aes(x= Axis.1, y=Axis.2, group=UserName), col = <span class="string">"darkgrey"</span>, alpha = <span class="number">0.6</span>) +</span><br><span class="line">  theme(panel.background = element_blank(),</span><br><span class="line">        panel.grid.major = element_blank(),</span><br><span class="line">        panel.grid.minor = element_blank(),</span><br><span class="line">        legend.title = element_blank(),</span><br><span class="line">        legend.text = element_text(size=<span class="number">9</span>),</span><br><span class="line">        legend.position = <span class="string">'bottom'</span>,</span><br><span class="line">        axis.text = element_text(size=<span class="number">4</span>),</span><br><span class="line">        axis.title = element_text(size=<span class="number">9</span>),</span><br><span class="line">         aspect.ratio = <span class="number">1</span>) +</span><br><span class="line">  guides(color = guide_legend(ncol = <span class="number">1</span>)) +</span><br><span class="line">  annotate(<span class="string">"text"</span>, x = <span class="number">0.33</span>, y = -<span class="number">0.33</span>, label = paste0(<span class="string">"p-value="</span>,pval), size = <span class="number">2</span>) +</span><br><span class="line">  xlab(paste0(<span class="string">"PC 1 ["</span>,percent_var[<span class="number">1</span>],<span class="string">"%]"</span>)) +</span><br><span class="line">  ylab(paste0(<span class="string">"PC 2 ["</span>,percent_var[<span class="number">2</span>],<span class="string">"%]"</span>)) </span><br><span class="line"></span><br><span class="line">fruit_micro + theme(legend.position = <span class="string">"none"</span>)</span><br><span class="line"></span><br><span class="line">fruit_leg &lt;- get_legend(fruit_micro)</span><br><span class="line"></span><br><span class="line"><span class="comment"># 合并四个图</span></span><br><span class="line"></span><br><span class="line">plotC_H &lt;- plot_grid(  food_micro + theme(legend.position = <span class="string">"none"</span>), </span><br><span class="line">          nutr_micro + theme(legend.position = <span class="string">"none"</span>), </span><br><span class="line">          grain_micro + theme(legend.position = <span class="string">"none"</span>), </span><br><span class="line">          fruit_micro + theme(legend.position = <span class="string">"none"</span>), </span><br><span class="line">          ncol = <span class="number">2</span>, </span><br><span class="line">          align = <span class="string">"h"</span>, </span><br><span class="line">          labels = c(<span class="string">"E"</span>, <span class="string">"F"</span>, <span class="string">"G"</span>, <span class="string">"H"</span>))</span><br><span class="line"></span><br><span class="line">save_plot(<span class="string">"Figure2E_H.pdf"</span>, plotC_H, base_width = <span class="number">3.5</span>, base_height = <span class="number">3.5</span>)</span><br></pre></td></tr></table></figure></p>
<p>5 | 出图，大功告成<br><img src="https://s2.ax1x.com/2019/06/15/VoQ8UA.png" alt="图三 普氏分析图"></p>
<h2 id="参考文章"><a href="#参考文章" class="headerlink" title="参考文章"></a>参考文章</h2><p>[1] <a href="https://www.cell.com/cell-host-microbe/fulltext/S1931-3128(19)30250-1" target="_blank" rel="noopener">https://www.cell.com/cell-host-microbe/fulltext/S1931-3128(19)30250-1</a></p>
  
	</div>
		
    <script type="text/javascript">
    var disqus_shortname = 'dayueban';
    var disqus_config = function () {
        this.page.url = 'https://dayueban.com/2019/06/15/procrustes分析/';
        this.page.identifier = '/2019/06/15/procrustes分析/';
        this.page.title = 'procrustes分析';
    };
    (function(){
      var d = document;
      var dsq = d.createElement('script');
      dsq.type = 'text/javascript';
      dsq.async = true;
      dsq.src = '//' + disqus_shortname + '.disqus.com/embed.js';
      (d.head || d.body).appendChild(dsq);
    })();
    </script>
   	       
	</article>
	
<nav class="article-nav clearfix">
 

<div class="next">
<a href="/2019/03/01/batchCorr包-非靶向代谢组学数据校正/" title="batchCorr包-非靶向代谢组学数据校正">
 <strong>下一篇：</strong><br> 
 <span>batchCorr包-非靶向代谢组学数据校正
</span>
</a>
</div>

</nav>

	
<section id="comment">
  
  <div id="disqus_thread">
    <noscript>Please enable JavaScript to view the <a href="//disqus.com/?ref_noscript">comments powered by Disqus.</a></noscript>
  </div>
  
</section>

</div>  
      <div class="openaside"><a class="navbutton" href="#" title="Show Sidebar"></a></div>

  <div id="toc" class="toc-aside">
  <strong class="toc-title">Contents</strong>
 
 <ol class="toc"><li class="toc-item toc-level-2"><a class="toc-link" href="#目的："><span class="toc-number">1.</span> <span class="toc-text">目的：</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#一、导入数据"><span class="toc-number">2.</span> <span class="toc-text">一、导入数据</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#二、进行procrustes分析并作图"><span class="toc-number">3.</span> <span class="toc-text">二、进行procrustes分析并作图</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#参考文章"><span class="toc-number">4.</span> <span class="toc-text">参考文章</span></a></li></ol>
 
  </div>

<div id="asidepart">
<div class="closeaside"><a class="closebutton" href="#" title="Hide Sidebar"></a></div>
<aside class="clearfix">

  
<div class="categorieslist">
	<p class="asidetitle">Categories</p>
		<ul>
		
		  
			<li><a href="/categories/Bioinformatics/" title="Bioinformatics">Bioinformatics<sup>3</sup></a></li>
		  
		
		  
			<li><a href="/categories/R-language/" title="R language">R language<sup>1</sup></a></li>
		  
		
		  
			<li><a href="/categories/代谢物/" title="代谢物">代谢物<sup>1</sup></a></li>
		  
		
		  
			<li><a href="/categories/代谢组学/" title="代谢组学">代谢组学<sup>11</sup></a></li>
		  
		
		  
			<li><a href="/categories/宏基因组/" title="宏基因组">宏基因组<sup>5</sup></a></li>
		  
		
		  
			<li><a href="/categories/机器学习/" title="机器学习">机器学习<sup>1</sup></a></li>
		  
		
		  
			<li><a href="/categories/胆汁酸/" title="胆汁酸">胆汁酸<sup>1</sup></a></li>
		  
		
		  
			<li><a href="/categories/软件下载/" title="软件下载">软件下载<sup>2</sup></a></li>
		  
		
		</ul>
</div>


  
<div class="tagslist">
	<p class="asidetitle">Tags</p>
		<ul class="clearfix">
		
			
				<li><a href="/tags/R/" title="R">R<sup>8</sup></a></li>
			
		
			
				<li><a href="/tags/数据分析/" title="数据分析">数据分析<sup>4</sup></a></li>
			
		
			
				<li><a href="/tags/beta多样性/" title="beta多样性">beta多样性<sup>2</sup></a></li>
			
		
			
				<li><a href="/tags/实验设计/" title="实验设计">实验设计<sup>2</sup></a></li>
			
		
			
				<li><a href="/tags/R包/" title="R包">R包<sup>2</sup></a></li>
			
		
			
				<li><a href="/tags/Statistics/" title="Statistics">Statistics<sup>1</sup></a></li>
			
		
			
				<li><a href="/tags/生化指标/" title="生化指标">生化指标<sup>1</sup></a></li>
			
		
			
				<li><a href="/tags/免疫/" title="免疫">免疫<sup>1</sup></a></li>
			
		
			
				<li><a href="/tags/菌群-宿主互作/" title="菌群-宿主互作">菌群-宿主互作<sup>1</sup></a></li>
			
		
			
				<li><a href="/tags/代谢组学/" title="代谢组学">代谢组学<sup>1</sup></a></li>
			
		
			
				<li><a href="/tags/ORF预测/" title="ORF预测">ORF预测<sup>1</sup></a></li>
			
		
			
				<li><a href="/tags/实用软件包/" title="实用软件包">实用软件包<sup>1</sup></a></li>
			
		
			
				<li><a href="/tags/芳香族氨基酸/" title="芳香族氨基酸">芳香族氨基酸<sup>1</sup></a></li>
			
		
			
				<li><a href="/tags/比较基因组学/" title="比较基因组学">比较基因组学<sup>1</sup></a></li>
			
		
			
				<li><a href="/tags/ggplot2/" title="ggplot2">ggplot2<sup>1</sup></a></li>
			
		
			
				<li><a href="/tags/面积图/" title="面积图">面积图<sup>1</sup></a></li>
			
		
			
				<li><a href="/tags/alpha/" title="alpha">alpha<sup>1</sup></a></li>
			
		
			
				<li><a href="/tags/data-analysis/" title="data analysis">data analysis<sup>1</sup></a></li>
			
		
			
				<li><a href="/tags/NMDS/" title="NMDS">NMDS<sup>1</sup></a></li>
			
		
			
				<li><a href="/tags/数据归一化/" title="数据归一化">数据归一化<sup>1</sup></a></li>
			
		
		</ul>
</div>


  <div class="linkslist">
  <p class="asidetitle">Links</p>
    <ul>
        
          <li>
            
            	<a href="http://www.codeceo.com/" target="_blank" title="一个面向程序员交流分享的新一代社区">码农网</a>
            
          </li>
        
          <li>
            
            	<a href="https://www.jianshu.com/u/1fda92d012f8" target="_blank" title="Dayueban(简书)">Dayueban(简书)</a>
            
          </li>
        
          <li>
            
            	<a href="http://www.sthda.com/english/" target="_blank" title="STHDA">STHDA</a>
            
          </li>
        
    </ul>
</div>

  
  <div class="archiveslist">
    <p class="asidetitle"><a href="/archives">Archives</a></p>
      <ul class="archive-list"><li class="archive-list-item"><a class="archive-list-link" href="/archives/2019/06/">June 2019</a><span class="archive-list-count">1</span></li><li class="archive-list-item"><a class="archive-list-link" href="/archives/2019/03/">March 2019</a><span class="archive-list-count">1</span></li><li class="archive-list-item"><a class="archive-list-link" href="/archives/2019/02/">February 2019</a><span class="archive-list-count">4</span></li><li class="archive-list-item"><a class="archive-list-link" href="/archives/2019/01/">January 2019</a><span class="archive-list-count">4</span></li><li class="archive-list-item"><a class="archive-list-link" href="/archives/2018/12/">December 2018</a><span class="archive-list-count">7</span></li><li class="archive-list-item"><a class="archive-list-link" href="/archives/2018/11/">November 2018</a><span class="archive-list-count">4</span></li><li class="archive-list-item"><a class="archive-list-link" href="/archives/2018/10/">October 2018</a><span class="archive-list-count">8</span></li></ul>
  </div>


  
  <div class="tagcloudlist">
    <p class="asidetitle">Tag Cloud</p>
    <div class="tagcloudlist clearfix">
       <a href="/tags/2018年终总结/" style="font-size: 10px;">2018年终总结</a> <a href="/tags/FXR受体/" style="font-size: 10px;">FXR受体</a> <a href="/tags/GIT/" style="font-size: 10px;">GIT</a> <a href="/tags/KEGG/" style="font-size: 10px;">KEGG</a> <a href="/tags/NADPH/" style="font-size: 10px;">NADPH</a> <a href="/tags/NMDS/" style="font-size: 10px;">NMDS</a> <a href="/tags/ORF预测/" style="font-size: 10px;">ORF预测</a> <a href="/tags/QC/" style="font-size: 10px;">QC</a> <a href="/tags/R/" style="font-size: 20px;">R</a> <a href="/tags/ROS/" style="font-size: 10px;">ROS</a> <a href="/tags/Regression/" style="font-size: 10px;">Regression</a> <a href="/tags/R包/" style="font-size: 13.33px;">R包</a> <a href="/tags/Statistics/" style="font-size: 10px;">Statistics</a> <a href="/tags/TGR5受体/" style="font-size: 10px;">TGR5受体</a> <a href="/tags/alpha/" style="font-size: 10px;">alpha</a> <a href="/tags/batchCorr/" style="font-size: 10px;">batchCorr</a> <a href="/tags/beta多样性/" style="font-size: 13.33px;">beta多样性</a> <a href="/tags/bioconductor/" style="font-size: 10px;">bioconductor</a> <a href="/tags/biomarker/" style="font-size: 10px;">biomarker</a> <a href="/tags/data-analysis/" style="font-size: 10px;">data analysis</a> <a href="/tags/ggplot2/" style="font-size: 10px;">ggplot2</a> <a href="/tags/informatics/" style="font-size: 10px;">informatics</a> <a href="/tags/statTarget/" style="font-size: 10px;">statTarget</a> <a href="/tags/代谢物靶标/" style="font-size: 10px;">代谢物靶标</a> <a href="/tags/代谢组学/" style="font-size: 10px;">代谢组学</a> <a href="/tags/使用软件包/" style="font-size: 10px;">使用软件包</a> <a href="/tags/免疫/" style="font-size: 10px;">免疫</a> <a href="/tags/单变量/" style="font-size: 10px;">单变量</a> <a href="/tags/单变量分析/" style="font-size: 10px;">单变量分析</a> <a href="/tags/变量选择/" style="font-size: 10px;">变量选择</a> <a href="/tags/多元统计/" style="font-size: 10px;">多元统计</a> <a href="/tags/多元统计分析/" style="font-size: 10px;">多元统计分析</a> <a href="/tags/实用软件包/" style="font-size: 10px;">实用软件包</a> <a href="/tags/实验设计/" style="font-size: 13.33px;">实验设计</a> <a href="/tags/富集分析/" style="font-size: 10px;">富集分析</a> <a href="/tags/支链氨基酸/" style="font-size: 10px;">支链氨基酸</a> <a href="/tags/数据分析/" style="font-size: 16.67px;">数据分析</a> <a href="/tags/数据归一化/" style="font-size: 10px;">数据归一化</a> <a href="/tags/普氏分析/" style="font-size: 10px;">普氏分析</a> <a href="/tags/机器学习/" style="font-size: 10px;">机器学习</a> <a href="/tags/杂谈/" style="font-size: 10px;">杂谈</a> <a href="/tags/比较基因组学/" style="font-size: 10px;">比较基因组学</a> <a href="/tags/液相色谱质谱/" style="font-size: 10px;">液相色谱质谱</a> <a href="/tags/生化指标/" style="font-size: 10px;">生化指标</a> <a href="/tags/粘膜屏障/" style="font-size: 10px;">粘膜屏障</a> <a href="/tags/糖代谢/" style="font-size: 10px;">糖代谢</a> <a href="/tags/肠道/" style="font-size: 10px;">肠道</a> <a href="/tags/肠道菌群定植/" style="font-size: 10px;">肠道菌群定植</a> <a href="/tags/芳香族氨基酸/" style="font-size: 10px;">芳香族氨基酸</a> <a href="/tags/菌群/" style="font-size: 10px;">菌群</a> <a href="/tags/菌群-宿主互作/" style="font-size: 10px;">菌群-宿主互作</a> <a href="/tags/面积图/" style="font-size: 10px;">面积图</a>
    </div>
  </div>


  <div class="rsspart">
	<a href="/atom.xml" target="_blank" title="rss">RSS</a>
</div>

</aside>
</div>
    </div>
    <footer><div id="footer">
	
	<div class="line">
		<span></span>
		<div class="author"></div>
	</div>
	
	
	<section class="info">
		<p> Hello ,I&#39;m Dayueban. <br>
			You have to work really hard to look effortless.</p>
	</section>
	 
	<div class="social-font">
		
		<a href="http://weibo.com/2689498473" target="_blank" class="icon-weibo" title="微博"></a>
		
		
		<a href="https://github.com/https://github.com/Dayueban" target="_blank" class="icon-github" title="github"></a>
		
		
		
		<a href="https://twitter.com/https://twitter.com/HeMaozhang" target="_blank" class="icon-twitter" title="twitter"></a>
		
		
		
		<a href="https://www.linkedin.com/in/https://www.linkedin.com/in/%E8%8C%82%E7%AB%A0-%E4%BD%95-481a59122/" target="_blank" class="icon-linkedin" title="linkedin"></a>
		
		
		<a href="https://www.douban.com/people/https://www.douban.com/people/146697967/" target="_blank" class="icon-douban" title="豆瓣"></a>
		
		
		<a href="http://www.zhihu.com/people/https://www.zhihu.com/people/he-mao-zhang/activities" target="_blank" class="icon-zhihu" title="知乎"></a>
		
		
		
		<a href="mailto:jnzd_hemaozhang@hotmail.com" target="_blank" class="icon-email" title="Email Me"></a>
		
	</div>
			
		

		<p class="copyright">
		Powered by <a href="http://hexo.io" target="_blank" title="hexo">hexo</a> and Theme by <a href="https://github.com/wuchong/jacman" target="_blank" title="Jacman">Jacman</a> © 2019 
		
		<a href="/about" target="_blank" title="Dayueban">Dayueban</a>
		
		
		</p>
</div>
</footer>
    <script src="/js/jquery-2.0.3.min.js"></script>
<script src="/js/jquery.imagesloaded.min.js"></script>
<script src="/js/gallery.js"></script>
<script src="/js/jquery.qrcode-0.12.0.min.js"></script>

<script type="text/javascript">
$(document).ready(function(){ 
  $('.navbar').click(function(){
    $('header nav').toggleClass('shownav');
  });
  var myWidth = 0;
  function getSize(){
    if( typeof( window.innerWidth ) == 'number' ) {
      myWidth = window.innerWidth;
    } else if( document.documentElement && document.documentElement.clientWidth) {
      myWidth = document.documentElement.clientWidth;
    };
  };
  var m = $('#main'),
      a = $('#asidepart'),
      c = $('.closeaside'),
      o = $('.openaside');
  c.click(function(){
    a.addClass('fadeOut').css('display', 'none');
    o.css('display', 'block').addClass('fadeIn');
    m.addClass('moveMain');
  });
  o.click(function(){
    o.css('display', 'none').removeClass('beforeFadeIn');
    a.css('display', 'block').removeClass('fadeOut').addClass('fadeIn');      
    m.removeClass('moveMain');
  });
  $(window).scroll(function(){
    o.css("top",Math.max(80,260-$(this).scrollTop()));
  });
  
  $(window).resize(function(){
    getSize(); 
    if (myWidth >= 1024) {
      $('header nav').removeClass('shownav');
    }else{
      m.removeClass('moveMain');
      a.css('display', 'block').removeClass('fadeOut');
      o.css('display', 'none');
      
      $('#toc.toc-aside').css('display', 'none');
        
    }
  });
});
</script>

<script type="text/javascript">
$(document).ready(function(){ 
  var ai = $('.article-content>iframe'),
      ae = $('.article-content>embed'),
      t  = $('#toc'),
      ta = $('#toc.toc-aside'),
      o  = $('.openaside'),
      c  = $('.closeaside');
  if(ai.length>0){
    ai.wrap('<div class="video-container" />');
  };
  if(ae.length>0){
   ae.wrap('<div class="video-container" />');
  };
  c.click(function(){
    ta.css('display', 'block').addClass('fadeIn');
  });
  o.click(function(){
    ta.css('display', 'none');
  });
  $(window).scroll(function(){
    ta.css("top",Math.max(140,320-$(this).scrollTop()));
  });
});
</script>


<script type="text/javascript">
$(document).ready(function(){ 
  var $this = $('.share'),
      url = $this.attr('data-url'),
      encodedUrl = encodeURIComponent(url),
      title = $this.attr('data-title'),
      tsina = $this.attr('data-tsina'),
      description = $this.attr('description');
  var html = [
  '<div class="hoverqrcode clearfix"></div>',
  '<a class="overlay" id="qrcode"></a>',
  '<a href="https://www.facebook.com/sharer.php?u=' + encodedUrl + '" class="article-share-facebook" target="_blank" title="Facebook"></a>',
  '<a href="https://twitter.com/intent/tweet?url=' + encodedUrl + '" class="article-share-twitter" target="_blank" title="Twitter"></a>',
  '<a href="#qrcode" class="article-share-qrcode" title="微信"></a>',
  '<a href="http://widget.renren.com/dialog/share?resourceUrl=' + encodedUrl + '&srcUrl=' + encodedUrl + '&title=' + title +'" class="article-share-renren" target="_blank" title="人人"></a>',
  '<a href="http://service.weibo.com/share/share.php?title='+title+'&url='+encodedUrl +'&ralateUid='+ tsina +'&searchPic=true&style=number' +'" class="article-share-weibo" target="_blank" title="微博"></a>',
  '<span title="Share to"></span>'
  ].join('');
  $this.append(html);

  $('.hoverqrcode').hide();

  var myWidth = 0;
  function updatehoverqrcode(){
    if( typeof( window.innerWidth ) == 'number' ) {
      myWidth = window.innerWidth;
    } else if( document.documentElement && document.documentElement.clientWidth) {
      myWidth = document.documentElement.clientWidth;
    };
    var qrsize = myWidth > 1024 ? 200:100;
    var options = {render: 'image', size: qrsize, fill: '#2ca6cb', text: url, radius: 0.5, quiet: 1};
    var p = $('.article-share-qrcode').position();
    $('.hoverqrcode').empty().css('width', qrsize).css('height', qrsize)
                          .css('left', p.left-qrsize/2+20).css('top', p.top-qrsize-10)
                          .qrcode(options);
  };
  $(window).resize(function(){
    $('.hoverqrcode').hide();
  });
  $('.article-share-qrcode').click(function(){
    updatehoverqrcode();
    $('.hoverqrcode').toggle();
  });
  $('.article-share-qrcode').hover(function(){}, function(){
      $('.hoverqrcode').hide();
  });
});   
</script>




<script type="text/javascript">

var disqus_shortname = 'dayueban';

(function(){
  var dsq = document.createElement('script');
  dsq.type = 'text/javascript';
  dsq.async = true;
  dsq.src = '//' + disqus_shortname + '.disqus.com/embed.js';
  (document.getElementsByTagName('head')[0] || document.getElementsByTagName('body')[0]).appendChild(dsq);
}());
(function(){
  var dsq = document.createElement('script');
  dsq.type = 'text/javascript';
  dsq.async = true;
  dsq.src = '//' + disqus_shortname + '.disqus.com/count.js';
  (document.getElementsByTagName('head')[0] || document.getElementsByTagName('body')[0]).appendChild(dsq);
}());
</script>








<link rel="stylesheet" href="/fancybox/jquery.fancybox.css" media="screen" type="text/css">
<script src="/fancybox/jquery.fancybox.pack.js"></script>
<script type="text/javascript">
$(document).ready(function(){ 
  $('.article-content').each(function(i){
    $(this).find('img').each(function(){
      if ($(this).parent().hasClass('fancybox')) return;
      var alt = this.alt;
      if (alt) $(this).after('<span class="caption">' + alt + '</span>');
      $(this).wrap('<a href="' + this.src + '" title="' + alt + '" class="fancybox"></a>');
    });
    $(this).find('.fancybox').each(function(){
      $(this).attr('rel', 'article' + i);
    });
  });
  if($.fancybox){
    $('.fancybox').fancybox();
  }
}); 
</script>



<!-- Analytics Begin -->



<script>
var _hmt = _hmt || [];
(function() {
  var hm = document.createElement("script");
  hm.src = "//hm.baidu.com/hm.js?e6d1f421bbc9962127a50488f9ed37d1";
  var s = document.getElementsByTagName("script")[0]; 
  s.parentNode.insertBefore(hm, s);
})();
</script>



<script type="text/javascript">var cnzz_protocol = (("https:" == document.location.protocol) ? " https://" : " http://");document.write(unescape("%3Cspan id='cnzz_stat_icon_1274989387'%3E%3C/span%3E%3Cscript src='" + cnzz_protocol + "s13.cnzz.com/z_stat.php%3Fid%3D1274989387%26show%3Dpic' type='text/javascript'%3E%3C/script%3E"));</script>

<!-- Analytics End -->

<!-- Totop Begin -->

	<div id="totop">
	<a title="Back to Top"><img src="/img/scrollup.png"></a>
	</div>
	<script src="/js/totop.js"></script>

<!-- Totop End -->

<!-- MathJax Begin -->
<!-- mathjax config similar to math.stackexchange -->


<!-- MathJax End -->

<!-- Tiny_search Begin -->

<script>
var option = {
  engineKey: ''
};
(function(w,d,t,u,n,s,e){
  s = d.createElement(t);
  s.src = u;
  s.async = 1;
  w[n] = function(r){
    w[n].opts = r;
  };
  e = d.getElementsByTagName(t)[0];
  e.parentNode.insertBefore(s, e);
})(window,document,'script','//tinysou-cdn.b0.upaiyun.com/ts.js','_ts');
_ts(option);
</script>

<!-- Tiny_search End -->

  </body>
</html>
